src/core/js/models/itemsModel.js

```diff
@@ -0,0 +1,52 @@
+define([
+    'core/js/adapt',
+    'core/js/models/componentModel'
+], function(Adapt, ComponentModel) {
+
+    var ItemsModel = ComponentModel.extend({
+
+        reset: function(type) {
+            this.resetItems();
+            ComponentModel.prototype.reset.call(this, type);
+        },
+
+        resetItems: function() {
+            _.each(this.get('_items'), function(item) {
+                item._isVisited = false;
+            });
+        },
+        
+        getItemCount: function() {
+            return this.get('_items').length;
+        },
+
+        getCurrentItem: function(index) {
```

Comment:
Since `index` is a parameter, would this be better called `getItemAtIndex`?

To me something called `getCurrentItem` should return the item using an internal indexer, not something that must be passed in.